evap/evaluation/tests/test_auth.py

```diff
@@ -30,25 +30,29 @@ def test_login_url_generation(self):
         self.assertEqual(reversed_url, '/key/{}'.format(self.external_user.login_key))
 
     def test_login_url_works(self):
-        self.assertRedirects(self.app.get(reverse("contributor:index")), "/?next=/contributor/")
-
         url_with_key = reverse("evaluation:login_key_authentication", args=[self.external_user.login_key])
-        page = self.app.get(url_with_key).follow().follow()
+        page = self.app.get(url_with_key)
+        self.assertContains(page, 'Login')
+        self.assertContains(page, self.external_user.full_name)
+
+        page = self.app.post(url_with_key).follow().follow()
```

Comment:
While this tests the typical login flow, it does not test whether the special case we are doing this for works (email client issues get, browser issues get, browser issues post).
I'd add
```
old_key = self.external_user.login_key
old_valid_until = self.external_user.login_key_valid_until
```
before the get and
```
self.external_user.refresh_from_db()
self.assertEqual(old_login_key, self.external_user.login_key)
self.assertEqual(old_login_key_valid_until, self.external_user.login_key_valid_until)
```
afterwards.

Alternatively, we could add another `self.app.get` between the `get` and the `post` calls, however, I think this might work just because the app remembers something about the user. Not sure about this, though (how much state does `self.app` have in the tests?)